### PR TITLE
Add event_track_ticket.

### PR DESCRIPTION
**Pull request by tschanzt**
This adds the event_track_ticket module. The goal of the module is to move the whole ticket machinery to the tracks. This allows us to easier create tickets if our tracks have different cost.

### CODE REVIEW

**Comment by tschanzt** [comment]
@simahawk you want to check on the docs since you know it as well?

File: event_track_ticket/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Event Track Ticket',
+    'description': 'Event Tickets which are controlled by Track.',
+    'version': '11.0.1.0.0',
+    'author': 'Camptocamp',
```
Comment (chienandalu) [inline event_track_ticket/__manifest__.py]: Missing author: "Odoo Community Association (OCA)"

File: event_track_ticket/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Event Track Ticket',
+    'description': 'Event Tickets which are controlled by Track.',
```
Comment (chienandalu) [inline event_track_ticket/__manifest__.py]: Use summary key instead

File: event_track_ticket/models/event_track.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+class Track(models.Model):
+    _inherit = 'event.track'
+
+    ticket_ids = fields.One2many(
+        comodel_name='event.event.ticket',
+        inverse_name='track_id',
+        string='Tickets'
+    )
+    # registration_ids = fields.Many2many(
```
Comment (chienandalu) [inline event_track_ticket/models/event_track.py]: Remove it if not necessary

File: setup/event_track_ticket/setup.cfg [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[bdist_wheel]
+universal=1
```
Comment (chienandalu) [inline setup/event_track_ticket/setup.cfg]: ?

File: event_track_ticket/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    'version': '11.0.1.0.0',
+    'author': 'Camptocamp',
+    'license': 'AGPL-3',
+    'category': 'Others',
+    'depends': [
+        'event',
```
Comment (chienandalu) [inline event_track_ticket/__manifest__.py]: `event_sale` already pulls from `event`

File: setup/event_track_ticket/odoo/addons/event_track_ticket [hunk 6]
```diff
@@ -0,0 +1 @@
+../../../../event_track_ticket/
```
Comment (chienandalu) [inline setup/event_track_ticket/odoo/addons/event_track_ticket]: ??

File: setup/event_track_ticket/event_track_ticket [hunk 7]
```diff
@@ -0,0 +1 @@
+event_track_ticket
```
Comment (chienandalu) [inline setup/event_track_ticket/event_track_ticket]: ??

**Review by chienandalu — CHANGES_REQUESTED** [review]
Just a formal review:

File: event_track_ticket/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+        'event_sale',
+        'website_event_track',
+    ],
+    'website': 'http://www.camptocamp.com',
```
Comment (simahawk) [inline event_track_ticket/__manifest__.py]: https://github.com/OCA/event

File: event_track_ticket/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    'data': [
+        'views/event_ticket.xml',
+        'views/event_views.xml',
+        'views/event_track.xml',
+    ],
+    'installable': True,
```
Comment (simahawk) [inline event_track_ticket/__manifest__.py]: you can skip this

File: event_track_ticket/readme/CONTRIBUTORS.txt [hunk 10]
```diff
@@ -0,0 +1,4 @@
+* Timon Tschanz <timon.tschanz@camptocamp.com>
```
Comment (simahawk) [inline event_track_ticket/readme/CONTRIBUTORS.txt]: TXT?

File: event_track_ticket/tests/test_registration.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo.tests.common import SavepointCase
+from odoo.exceptions import ValidationError
+
+
+# TODO: update tests
```
Comment (simahawk) [inline event_track_ticket/tests/test_registration.py]: to update?

File: setup/event_track_ticket/.eggs/README.txt [hunk 12]
```diff
@@ -0,0 +1,6 @@
+This directory contains eggs that were downloaded by setuptools to build, test, and run plug-ins.
```
Comment (simahawk) [inline setup/event_track_ticket/.eggs/README.txt]: @tschanzt pls remove `setup` folder: it won't hurt as it's going to be re-generated but is pointless for reviewing

**Review by simahawk — CHANGES_REQUESTED** [review]

File: event_track_ticket/tests/test_registration.py (hunk 11 above)
Comment (tschanzt) [reply event_track_ticket/tests/test_registration.py]: you'll have to ask the guy who wrote it :smile:.Im not sure the main difference is that the seats are handled on the track. which is checked in test_track but registrations still should work the same way.

**Comment by tschanzt** [comment]
@chienandalu @simahawk i updated the pr.

**Comment by tschanzt** [comment]
@simahawk i updated the pr with our new developments.

**Comment by tschanzt** [comment]
@simahawk ping

**Review by simahawk — APPROVED** [review]
LG now.
@chienandalu all your comments seem addressed.
I did latest cleanup, generated readme, rebased and squashed.

**Review by chienandalu — APPROVED** [review]

**Comment by pedrobaeza** [comment]
As this depends on `event_sale`, this should be named `event_sale_track`, (and no need IMO of the word `ticket`, as `event_sale` adds the concept of ticket).

**Comment by rafaelbn** [comment]
Hi @tschanzt , please remember last comment, rename event_sale_track please! Thanks!

**Comment by tschanzt** [comment]
@pedrobaeza @rafaelbn i renamed the module.

**Comment by pedrobaeza** [comment]
Thanks for that. I'm not able now to review it now, but for sure other contributors can do it.

**Pull request merged**